src/components/MessagesList/MessagesGroup/Message/MessagePart/MessageBody.vue

```diff
@@ -245,9 +244,15 @@ export default {
 		},
 	},
 
-	setup() {
+	setup(props) {
+		const {
+			isEditable,
+			isFileShare,
+		} = useMessageInfo(props.token, props.id)
```

Comment:
`props.token` and  `props.id` are not needed to be reactive, they only have one value each during the lifetime of a component `MessageBody` in particular.

Reply:
> `props.token` and `props.id` are not needed to be reactive, they only have one value each during the lifetime of a component `MessageBody` in particular.

Is it a hard requirement of a component, e.g., it is unusable with update and must be recreated?

Anyway, it is a good practice to not lose reactivity even if it is not needed in a specific case.

Reply:
>  it is unusable with update and must be recreated?

Yes, as:
* If token has changed, MessagesList will be re-mounted
* If id has changed, Message component ( where MessageBody is used) will be re-mounted.

I thought it's a good practice not to abuse reactivity :p

Reply:
> * If token has changed, MessagesList will be re-mounted

This is not a requirement of `Message.vue` usage or its restriction. This is only its today's usage in another component.

Reply:
And it increases the chances that these lines will be copy/pasted or used as an example for another component with probably another composable, where it is important to not lose reactivity.

Reply:
If you are worried about performance - there is almost zero performance impact.

`toRef` doesn't create "new reactivity" here. We just providing a reference to `props.token` instead of providing the value of `props.token. It creates a small object with a `value` property to provide getter.

Reply:
I made them reactive